docs/language/functions.rst

```diff
@@ -353,6 +353,35 @@ values. Here is an example of an overloaded function:
 In the function foo, abs() is called with an ``int64`` so the second implementation
 of the function abs() is called.
 
+.. note::
+  The substrate target runtime requires function names to be unique.
+  Overloaded function names will be mangled in the ABI.
+  The function name will be concatenated with all of its argument types, separated by underscores.
+  Additionally, these rules apply:
+  - Struct types are represented by their field types (preceded with an extra underscore).
+  - Enum types are represented as their underlying uint8 type.
+  - Array types are recognizable by having ``Array`` appended. 
+  - Fixed size arrays will additionally have their length appended as well.
+
+  The following example illustrates some overloaded functions and their mangled name:
+
+  .. code-block:: solidity
+    
+    enum E { v1, v2 }
+    struct S { int256 i; bool b; address a; }
+
+    contract C {
+        // foo_
+        function foo() public pure {}
+
+        // foo_uint256_addressArray2Array
```

Comment:
Have you tested naming it as `foo_uint256_Address[2][]`? It appears clearer when we use square brackets for arrays.

Reply:
`foo_uint256_Address[2][]` is not an identifier